### DISCUSSION TITLE
Stream Problem

### DISCUSSION THREAD

**Discussion by zdh3303** [microsoft/msquic#2081] [Q&A] [answered]
hi，how can I create Stream in the server size once  the client connect to the server，and the server send data to the client(the client only receives the data send by the server). I call  StreamOpen  method in the server side, however, the client does not receive any message about the stream

**Comment by AvatarGanymede** [comment, marked as answer]
I meet the same problem. The server calls `StreamStart()`, but on the client side, the `QUIC_CONNECTION_EVENT_PEER_STREAM_STARTED` event is not triggered at all.

**Reply by nibanks** [reply]
Are you configuring the settings to specify the number of peer streams you allow/support?

**Reply by AvatarGanymede** [reply]
I set `Settings.PeerBidiStreamCount = 4;`, but still nothing happened 😌

**Reply by AvatarGanymede** [reply]
At event `QUIC_STREAM_EVENT_START_COMPLETE`, `Event->START_COMPLETE.PeerAccepted` field is empty. I guess this might be fatal, but I don't understand why.
The doc says:
> The QUIC_STREAM_START_FLAG_INDICATE_PEER_ACCEPT flag can be used to get the QUIC_STREAM_EVENT_PEER_ACCEPTED event to know when the stream becomes unblocked by flow control. If the peer already provided enough flow control to accept the stream when it was initially started, the QUIC_STREAM_EVENT_PEER_ACCEPTED event is not delivered and the QUIC_STREAM_EVENT_START_COMPLETE's PeerAccepted field will be TRUE. If is not initially accepted, if/once the peer provides enough flow control to allow the stream to be sent on the wire, then the QUIC_STREAM_EVENT_PEER_ACCEPTED event will be indicated to the app.

Maybe "the peer doesn't provide enough flow control to ...", but why?

**Reply by nibanks** [reply]
Just to make sure, you did start the connection, right? Did the connection get the CONNECTED event? Beyond that, we'll likely need logs to diagnose further ([instructions](https://github.com/microsoft/msquic/blob/main/docs/Diagnostics.md#trace-collection)).

**Reply by AvatarGanymede** [reply]
I SOLVE IT! The stream must send a datagram to the peer, then the peer will be notified.

**Reply by AvatarGanymede** [reply]
Thanks for your reply!

**Reply by nibanks** [reply]
Oh, you were just opening/starting the stream, and not sending anything? If you want that to indicate the stream to the peer (without sending), use the `QUIC_STREAM_START_FLAG_IMMEDIATE` stream start flag.

**Reply by nibanks** [reply]
Generally, apps don't want/need to indicate the stream immediately to the peer just because it was opened. So, MsQuic (by default) waits for application data to be queued before actually indicating anything to the peer. But that flag allows you to force the stream to be sent.

**Reply by AvatarGanymede** [reply]
I totally understand. Sincerely thanks!
